In terms of using clang.exe/clang++.exe on Windows and CMake choking when we use it:

```
cmake -G"Ninja"^
  -DCMAKE_C_COMPILER="C:/Users/hughb/Documents/GitHub/swift/build-msvc/Ninja-DebugAssert/llvm-windows-x86_64/bin/clang.exe"^
  -DCMAKE_CXX_COMPILER="C:/Users/hughb/Documents/GitHub/swift/build-msvc/Ninja-DebugAssert/llvm-windows-x86_64/bin/clang++.exe"
```

emits the following CMake generation errors
```
-- The C compiler identification is Clang 4.0.0
-- The CXX compiler identification is Clang 4.0.0
-- Check for working C compiler: C:/Users/hughb/Documents/GitHub/swift/build-msvc/Ninja-DebugAssert/llvm-windows-x86_64/bin/clang.exe
-- Check for working C compiler: C:/Users/hughb/Documents/GitHub/swift/build-msvc/Ninja-DebugAssert/llvm-windows-x86_64/bin/clang.exe -- broken
CMake Error at C:/Program Files/CMake/share/cmake-3.6/Modules/CMakeTestCCompiler.cmake:61 (message):
  The C compiler
  "C:/Users/hughb/Documents/GitHub/swift/build-msvc/Ninja-DebugAssert/llvm-windows-x86_64/bin/clang.exe"
  is not able to compile a simple test program.

  It fails with the following output:

   Change Dir: C:/Users/hughb/Desktop/stuff/CMakeFiles/CMakeTmp



  Run Build Command:"C:/Users/hughb/Documents/Utilities/ninja.exe"
  "cmTC_fe176"

  [1/2] Building C object CMakeFiles\cmTC_fe176.dir\testCCompiler.c.obj

  FAILED: CMakeFiles/cmTC_fe176.dir/testCCompiler.c.obj


  C:\Users\hughb\Documents\GitHub\swift\build-msvc\Ninja-DebugAssert\llvm-windows-x86_64\bin\clang.exe
  /nologo /DWIN32 /D_WINDOWS /W3 /D_DEBUG /MDd /Zi /Ob0 /Od /RTC1
  /showIncludes /FoCMakeFiles\cmTC_fe176.dir\testCCompiler.c.obj
  /FdCMakeFiles\cmTC_fe176.dir\ -c testCCompiler.c

  clang.exe: error: no such file or directory: '/nologo'

  clang.exe: error: no such file or directory: '/DWIN32'

  clang.exe: error: no such file or directory: '/D_WINDOWS'

  clang.exe: error: no such file or directory: '/W3'

  clang.exe: error: no such file or directory: '/D_DEBUG'

  clang.exe: error: no such file or directory: '/MDd'

  clang.exe: error: no such file or directory: '/Zi'

  clang.exe: error: no such file or directory: '/Ob0'

  clang.exe: error: no such file or directory: '/Od'

  clang.exe: error: no such file or directory: '/RTC1'

  clang.exe: error: no such file or directory: '/showIncludes'

  clang.exe: error: no such file or directory:
  '/FoCMakeFiles\cmTC_fe176.dir\testCCompiler.c.obj'

  clang.exe: error: no such file or directory:
  '/FdCMakeFiles\cmTC_fe176.dir\'

  ninja: build stopped: subcommand failed.





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt


-- Configuring incomplete, errors occurred!
See also "C:/Users/hughb/Desktop/stuff/CMakeFiles/CMakeOutput.log".
See also "C:/Users/hughb/Desktop/stuff/CMakeFiles/CMakeError.log".
```

--- a/utils/swift_build_support/swift_build_support/targets.py
+++ b/utils/swift_build_support/swift_build_support/targets.py
@@ -126,6 +126,8 @@ class StdlibDeploymentTarget(object):
 
     Android = Platform("android", archs=["armv7"])
 
+    Windows = Platform("windows", archs=["x86_64"])
+
     # The list of known platforms.
     known_platforms = [
         OSX,
@@ -135,7 +137,8 @@ class StdlibDeploymentTarget(object):
         Linux,
         FreeBSD,
         Cygwin,
-        Android]
+        Android,
+        Windows]
 
     # Cache of targets by name.
     _targets_by_name = dict((target.name, target)
@@ -180,6 +183,10 @@ class StdlibDeploymentTarget(object):
         elif system == 'CYGWIN_NT-10.0':
             if machine == 'x86_64':
                 return StdlibDeploymentTarget.Cygwin.x86_64
+
+        elif system == 'Windows':
+            if machine == "AMD64":
+                return StdlibDeploymentTarget.Windows.x86_64
 
         raise NotImplementedError('System "%s" with architecture "%s" is not '
                                   'supported' % (system, machine))

--- a/utils/swift_build_support/swift_build_support/toolchain.py
+++ b/utils/swift_build_support/swift_build_support/toolchain.py
@@ -16,7 +16,6 @@ Represent toolchain - the versioned executables.
 
 from __future__ import absolute_import
 
-import os
 import platform
 
 from . import cache_util
@@ -46,8 +45,13 @@ def _register(name, *tool):
     setattr(Toolchain, name, cache_util.reify(_getter))
 
 
-_register("cc", "clang")
-_register("cxx", "clang++")
+if platform.system() == 'Windows':
+    _register("cc", "clang-cl")
+    _register("cxx", "clang-cl")
+else:
+    _register("cc", "clang")
+    _register("cxx", "clang++")
+
 _register("ninja", "ninja", "ninja-build")
 _register("cmake", "cmake")
 _register("distcc", "distcc")

--- a/utils/swift_build_support/swift_build_support/toolchain.py
+++ b/utils/swift_build_support/swift_build_support/toolchain.py
@@ -16,6 +16,7 @@ Represent toolchain - the versioned executables.
 
 from __future__ import absolute_import
 
+import os
 import platform
 
 from . import cache_util
@@ -173,6 +174,15 @@ class Cygwin(Linux):
     pass
 
 
+class Windows(Toolchain):
+    def find_tool(self, *names):
+        for name in names:
+            found = which(name)
+            if found is not None:
+                return found
+        return None
+
+
 def host_toolchain(**kwargs):
     sys = platform.system()
     if sys == 'Darwin':
@@ -183,6 +193,8 @@ def host_toolchain(**kwargs):
         return FreeBSD()
     elif sys.startswith('CYGWIN'):
         return Cygwin()
+    elif sys == 'Windows':
+        return Windows()
     else:
         raise NotImplementedError('The platform "%s" does not have a defined '
                                   'toolchain.' % sys)

